### DISCUSSION TITLE
Question about Espresso write_input() implementation

### DISCUSSION THREAD

**Discussion by hironori-kondo** [Quantum-Accelerators/quacc#2341] [General]
In `espresso.py`, I see that `write_input()` has different functionality based on the choice of binary. For example, the phonon package relies on `write_espresso_ph()` as written in ASE. 

Why does the `pw.x` binary go to ASE's `write()` instead of its Espresso/pw-specific `write_espresso_in()` in the `io/espresso.py` file, where `write_espresso_ph()` is?

**Comment by Andrew-S-Rosen** [comment]
@nerdwarrior: Thank you for your question. 

Admittedly, I am not sure! I am not an Espresso user myself, and the original implementation was made by @tomdemeyere. 

If you think there are advantages for using `write_espresso_in`, please feel to propose a change. That said, does ASE's `write()` not default to `write_espresso_in()`?

**Reply by tomdemeyere** [reply]
ASE write will use write_espresso_in, I would say that it does not really matter? The espresso ph is not supported by the write function, we have to call it directly

**Reply by hironori-kondo** [reply]
I missed that---makes sense! Thank you!

**Reply by tomdemeyere** [reply]
No worries, if you have any other question about the espresso interface do not hesitate to ask!
